manager/manager.go

```diff
@@ -44,18 +44,26 @@ import (
 	"github.com/google/cadvisor/utils/sysfs"
 	"github.com/google/cadvisor/version"
 	"github.com/google/cadvisor/watcher"
+	"github.com/google/cadvisor/wss"
 
 	"github.com/opencontainers/runc/libcontainer/cgroups"
 	"k8s.io/klog/v2"
 	"k8s.io/utils/clock"
 )
 
+const (
+	nvidiaMetricsProviderKey = "nvidia"
+	perfMetricsProviderKey   = "perf"
+	wssMetricsProviderKey    = "wss"
+)
+
 var globalHousekeepingInterval = flag.Duration("global_housekeeping_interval", 1*time.Minute, "Interval between global housekeepings")
 var updateMachineInfoInterval = flag.Duration("update_machine_info_interval", 5*time.Minute, "Interval between machine info updates.")
 var logCadvisorUsage = flag.Bool("log_cadvisor_usage", false, "Whether to log the usage of the cAdvisor container")
 var eventStorageAgeLimit = flag.String("event_storage_age_limit", "default=24h", "Max length of time for which to store events (per type). Value is a comma separated list of key values, where the keys are event types (e.g.: creation, oom) or \"default\" and the value is a duration. Default is applied to all non-specified event types")
 var eventStorageEventLimit = flag.String("event_storage_event_limit", "default=100000", "Max number of events to store (per type). Value is a comma separated list of key values, where the keys are event types (e.g.: creation, oom) or \"default\" and the value is an integer. Default is applied to all non-specified event types")
 var applicationMetricsCountLimit = flag.Int("application_metrics_count_limit", 100, "Max number of application metrics to store (per container)")
+var wssResetInterval = flag.Uint64("wss_reset_interval", 5, "Reset interval for working set size metric, number of measurement cycles after which referenced bytes are cleared")
```

Comment:
Does "cycle" == "housekeeping interval" here?  Would it be more intuitive to measure in seconds?  Or even better, could we just reset it every housekeeping interval, and not have a knob at all?

Reply:
It could be good to have a mechanism to set a different interval for setting clear_refs than housekeeping interval. Sometimes it is better to measure wss in longer intervals. I've found following information:

> What you intend to use WSS for will direct how you measure it. Consider these three scenarios:
A) You are sizing main memory for an application, with the intent to keep it from paging (swapping). WSS will be measured in bytes over a long interval, say, one minute.
B) You are optimizing CPU caches. WSS will be measured in unique cachelines accessed over a short interval, say, one second or less. The cacheline size is architecture dependent, usually 64 bytes.
C) You are optimizing the TLB caches (translation lookaside buffer: a memory management unit cache for virtual to physical translation). WSS will be measured in unique pages accessed over a short interval, say, one second or less. The pagesize is architecture and OS configuration dependent, 4 Kbytes is commonly used.

It comes from http://www.brendangregg.com/wss.html#Estimation

If I measure time for setting clear_refs in seconds I need to introduce a new timer. Do you think that is a good idea?

Reply:
Actually, does it even make sense to have this differ from the housekeeping interval?  If my HK interval is 10 seconds, and wss reset is 6, won't I get higher and higher measurements each housekeeping until it is reset?  It seems like we would want to collect at the same interval that we reset at.

I guess this begs the question of whether this is meant to be used just for benchmarking, or whether this is something we expect to be enabled for production monitoring.  The warnings everywhere make this seem like mostly a benchmarking tool, in which case we could just tell users to use the housekeeping interval to tune the wss reset interval.  But if it is important to support different collection and rss reset intervals, then we would have to add this config.

Thinking about it more, I do agree that basing this off the HK interval is probably better than a separate timer.  We would need to reimplement jitter to prevent spikes, and it is good to be able to collect immediately before resetting, and NOT collect right after a reset.

Reply:
It is useful to observe how reference bytes growth in time and observe if we get only refernced bytes or it is a working set size (memory needed to keep working),  see: 
![image](https://user-images.githubusercontent.com/16666942/79740117-72a66400-82ff-11ea-95ba-ec88b2864864.png)

Summing up, I think that it is reasonable to introduce a new parameter which depends on house keeping interval.

Reply:
sgtm